### PR TITLE
Feedback tests 

### PR DESCRIPTION
**Pull request by SvetilnikST**

### CODE REVIEW

File: src/main/java/innowise/zuevsky/helpdesk/security/JwtTokenFilter.java [hunk 1]
```diff
@@ -18,30 +21,24 @@
...
-    chain.doFilter(request, response);
-  }
+	@Override
+	public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+			throws IOException, ServletException {
+		String token = jwtTokenProvider.resolveToken((HttpServletRequest) request);
+		try {
+			if (token != null && jwtTokenProvider.validateToken(token)) {
+				Authentication authentication = jwtTokenProvider.getAuthentication(token);
+				if (authentication != null) {
```
Comment (kislyakdmitry) [inline src/main/java/innowise/zuevsky/helpdesk/security/JwtTokenFilter.java:33]: if null?

File: src/main/java/innowise/zuevsky/helpdesk/service/FeedbackService.java [hunk 2]
```diff
@@ -1,66 +1,42 @@
...
-
-    return true;
-  }
+	private final FeedbackRepository feedbackRepository;
+	private final FeedbackMapper feedbackMapper;
+	private final TicketsService ticketsService;
+
+	public FeedbackDto getFeedbackById(Long feedbackId) {
+		return feedbackRepository.findById(feedbackId).map(feedbackMapper::mapFeedbackToFeedbackDto)
+				.orElseThrow(() -> new FeedbackNotFoundException(feedbackId));
```
Comment (kislyakdmitry) [inline src/main/java/innowise/zuevsky/helpdesk/service/FeedbackService.java:22]: формат

File: src/test/java/innowise/zuevsky/helpdesk/service/TicketsServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+		verify(ticketsRepository).findById(anyLong());
+	}
+
+	@Test
+	void validateTicketOwnerBelongUser_shouldThrow() {
+		// when
+		when(ticketsRepository.findById(anyLong())).thenReturn(Optional.of(ticket));
+		when(ticketMapper.mapTicketInTicketDto(ticket)).thenReturn(ticketDto);
+
+		TicketOwnerNotBelongsToUserException exception = assertThrows(TicketOwnerNotBelongsToUserException.class,
```
Comment (kislyakdmitry) [inline src/test/java/innowise/zuevsky/helpdesk/service/TicketsServiceTest.java:60]: [](https://www.baeldung.com/assertj-exception-assertion#1-using-assertthatthrownby)3.1. Using assertThatThrownBy()
Let's check if indexing an out of bounds item in a list raises an IndexOutOfBoundsException:

assertThatThrownBy(() -> {
    List<String> list = Arrays.asList("String one", "String two");
    list.get(2);
}).isInstanceOf(IndexOutOfBoundsException.class)
  .hasMessageContaining("Index: 2, Size: 2");
Notice how the code fragment that might throw an exception gets passed as a lambda expression.

Of course, we can leverage various standard AssertJ assertions here like:

.hasMessage("Index: %s, Size: %s", 2, 2)
.hasMessageStartingWith("Index: 2")
.hasMessageContaining("2")
.hasMessageEndingWith("Size: 2")
.hasMessageMatching("Index: \\d+, Size: \\d+")
.hasCauseInstanceOf(IOException.class)
.hasStackTraceContaining("java.io.IOException");

**Review by kislyakdmitry — CHANGES_REQUESTED** [review]

File: src/main/java/innowise/zuevsky/helpdesk/dto/FeedbackSaveDto.java [hunk 4]
```diff
@@ -11,13 +13,21 @@
 @NoArgsConstructor
 @Builder
 public class FeedbackSaveDto {
-  @Min(value = 1, message = "Rate should not be less than 1")
-  @Max(value = 5, message = "Rate should not be greater than 5")
-  private Integer rate;
 
-  private String text;
+	@ApiModelProperty(notes = "Rate in feedback from 1 to 5", example = "5")
```
[the remark this answers is not in the document]
Comment (SvetilnikST) [reply src/main/java/innowise/zuevsky/helpdesk/dto/FeedbackSaveDto.java:17]: fix

File: src/main/java/innowise/zuevsky/helpdesk/security/JwtTokenFilter.java (hunk 1 above)
Comment (SvetilnikST) [reply src/main/java/innowise/zuevsky/helpdesk/security/JwtTokenFilter.java:33]: fix

File: src/main/java/innowise/zuevsky/helpdesk/service/FeedbackService.java (hunk 2 above)
Comment (SvetilnikST) [reply src/main/java/innowise/zuevsky/helpdesk/service/FeedbackService.java]: fix

**Review by kislyakdmitry — APPROVED** [review]

**Pull request merged**